src/OpenTelemetry.Extensions/Trace/ActivitySpec.cs

```diff
@@ -0,0 +1,134 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Collections.ObjectModel;
+using System.Diagnostics;
+using System.Globalization;
+using System.Text.Encodings.Web;
+using System.Text.Json;
+
+namespace OpenTelemetry.Extensions.Trace;
+
+/// <summary>
+/// Activity per spec.
+/// </summary>
+public class ActivitySpec
+{
+    private static readonly JsonSerializerOptions JsonSerializerOptions = new()
+    {
+        PropertyNamingPolicy = new SnakeCaseNamingPolicy(),
+        WriteIndented = true,
+        Encoder = JavaScriptEncoder.UnsafeRelaxedJsonEscaping,
+    };
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="ActivitySpec"/> class.
+    /// Creates an instance of <see cref="ActivitySpec"/> from an <see cref="Activity"/>.
+    /// </summary>
+    /// <param name="activity">Activity.</param>
+    /// <param name="signal">Signal for which mapping is called.</param>
+    public ActivitySpec(Activity activity, Signal signal)
+    {
+#if NET
+        ArgumentNullException.ThrowIfNull(activity);
+#else
+        if (activity == null)
+        {
+            throw new ArgumentOutOfRangeException(nameof(activity));
+        }
+#endif
+
+        this.Name = activity.DisplayName;
+        this.ActivitySpecContext = new ActivitySpecContext(activity.Context);
+        this.ParentId = activity.ParentSpanId == default
+            ? string.Empty
+            : activity.ParentSpanId.ToString();
+        this.StartTime = FormatTimestamp(activity.StartTimeUtc);
+        this.EndTime = signal == Signal.Heartbeat
+            ? string.Empty
+            : FormatTimestamp(activity.StartTimeUtc.Add(activity.Duration));
+        this.StatusCode = activity.Status.ToString();
```

Comment:
Do we need a default value here if Status was [Unset](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.activitystatuscode?view=net-9.0)?

Reply:
i am in the middle of transitioning of serialization from https://opentelemetry.io/docs/concepts/signals/traces/#spans to https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/trace/v1/trace.proto. instead of just sending a span i'd send everything like it is in proto spec but it will be in json. this will be fixed.